clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1495,13 +1535,64 @@ public long position(TopicPartition partition) {
                 throw new IllegalArgumentException("You can only check the position for partitions assigned to this consumer.");
             Long offset = this.subscriptions.position(partition);
             while (offset == null) {
+                updateFetchPositions(10);
```

Comment:
Why don't we just call `position(partition, Duration.ofMillis(Long.MAX_VALUE)`?

Reply:
Not sure how important this is, but `Long.MAX_VALUE` isn't exactly the current behavior. It would only be a difference if someone is actually waiting for more than 24 days for the call to complete. This probably doesn't happen, but in my PR, I opted to just keep the exact current behavior of blocking forever.

That said, I'm not sure you meant to drop down from `Long.MAX_VALUE` and `retryBackoffMs` in the old code to 10 ms here.